helix-core/src/textobject.rs

```diff
@@ -198,6 +200,83 @@ pub fn textobject_paragraph(
     Range::new(anchor, head)
 }
 
+pub fn textobject_indentation_level(
+    slice: RopeSlice,
+    range: Range,
+    count: usize,
+    indent_width: usize,
+    tab_width: usize,
+) -> Range {
+    let (mut line_start, mut line_end) = range.line_range(slice);
+    let mut min_indent: Option<usize> = None;
+
+    // Find the innermost indent represented by the current selection range.
+    // Range could be only on one line, so we need an inclusive range in the
+    // loop definition.
+    for i in line_start..=line_end {
+        let line = slice.line(i);
+
+        // Including empty lines leads to pathological behaviour, where having
+        // an empty line in a multi-line selection causes the entire buffer to
+        // be selected, which is not intuitively what we want.
+        if !rope_is_line_ending(line) {
+            let indent_level = indent_level_for_line(line, tab_width, indent_width);
+            min_indent = if let Some(prev_min_indent) = min_indent {
+                Some(cmp::min(indent_level, prev_min_indent))
+            } else {
+                Some(indent_level)
+            }
+        }
+    }
+
+    // It can happen that the selection consists of an empty line, so min_indent
+    // will be untouched, in which case we can skip the rest of the function
+    // and no-op.
+    if min_indent.is_none() {
+        return range;
+    }
+
+    min_indent = Some(min_indent.unwrap() + 1 - count);
```

Comment:
This could be a redefinition of `min_indent` as a number since from this point on it is always Some. That removes any need for unwrap below.